packages/backend/src/apps/m365-excel/common/interceptors/request-error-handler.ts

```diff
@@ -11,6 +11,26 @@ type ThrowingHandler = (
 // Handle MS rate limiting us
 //
 const handle429: ThrowingHandler = ($, error) => {
+  // Edge case: Thus far, _only_ 429s from the Excel endpoint are retriable
+  // because Microsoft applies dynamic rate limits for Excel, and we've verified
+  // with GovTech that these 429s have no impact on other M365 users.
+  //
+  // https://learn.microsoft.com/en-us/graph/workbook-best-practice?tabs=http#reduce-throttling-errors
+  //
+  // Excel endpoints are uniquely identified by the `/workbook/` url segment.
+  // It's safe to directly check for this substring in the URL because '/'s are
+  // escaped if they're from user input.
+  //
+  // FIXME (ogp-weeloong): eval if we can remove this and just retry _all_ 429s
+  // once we get bullmq pro in.
+  if (error.response.config.url.includes('/workbook/')) {
+    const retryAfterMs = Number(error.response?.headers?.['retry-after']) * 1000
```

Comment:
What's a typical value of the retry-after header? Is there a chance that the retries compound and worsens the situation?

Reply:
Good question! We block queries to excel when they're above our very low rate limit, so we're not thundering-herding Excel and thus not compounding the problem.

The only issue we have is that our queue will drain slowly, but users are aware of this.  This PR further reduces our excel rate limit to further lower the odds of having a excel 429.